scipy/optimize/_differentialevolution.py

```diff
@@ -480,41 +622,85 @@ def solve(self):
-            for candidate in range(np.size(self.population, 0)):
+                
+            # brake the population to subpopulation depending on the size of the pool
+            # determine number of sub-populations 'nsp', or number of 
+            # parallel evaluations, which is 'nsp + 1'
+            # and the length of the reminder 'lenrem'
+            nsp, lenrem = divmod(np.size(self.population, 0), self.poolsize)
+            
+            itsp = 0
+            # iterate among sub-populations
+            # when self.poolsize == 1 the mutation is aggressive
+            # when self.poolsize > 1 the mutation is quasi-aggressive
+            while itsp <= nsp:
+                # determine the length of the subpopulation
+                lensp = self.poolsize if (itsp < nsp) else lenrem
+
+                # initialize list for all members (parameters) of the current subpopulation
+                spparams = []
+                # mutate parameters for the sub-population
+                for candidate in range(lensp):
+                    trial = self._mutate(candidate + self.poolsize*itsp)
+                    self._ensure_constraint(trial)
+                    spparams.append(self._scale_parameters(trial))
+
+                    nfev += 1
+
+                # in parallel case the self.func must return a list of energies
+                # for the whole subpopulation
+                if len(self.args) is not 0:
+                    spenergies = self.pool_map(lambda pars: self.func(pars, *self.args), spparams)
+                else:
+                    spenergies = self.pool_map(self.func, spparams)
+                    
+                # check the number of evaluation of the functions
+                # (this, perhaps, should be deprecated as unnecessary)
                 if nfev > self.maxfun:
                     warning_flag = True
                     status_message = _status_message['maxfev']
                     break
 
-                trial = self._mutate(candidate)
-                self._ensure_constraint(trial)
-                parameters = self._scale_parameters(trial)
-
-                energy = self.func(parameters, *self.args)
-                nfev += 1
-
-                if energy < self.population_energies[candidate]:
-                    self.population[candidate] = trial
-                    self.population_energies[candidate] = energy
-
-                    if energy < self.population_energies[0]:
-                        self.population_energies[0] = energy
-                        self.population[0] = trial
-
-            # stop when the fractional s.d. of the population is less than tol
-            # of the mean energy
-            convergence = (np.std(self.population_energies) /
-                           np.abs(np.mean(self.population_energies) +
-                                  _MACHEPS))
-
+                # update population and their energies if subpopulation members are 
```

Comment:
Should be able to replace the for loop in 670-689 by:

    # find out which trial candidates have have lower energy than the existing population and replace the original population members
    offset = self.poolsize*itsp
    self.population[offset: offset + lensp] = np.where(spenergies < self.population_energies[offset: offset + lensp], trials, self.population[offset: offset + lensp])

    # also replace the energies if they got lower
    self.population_energies[offset: offset + lensp] = np.where(spenergies < self.population_energies[offset: offset + lensp], spenergies, self.population_energies[offset: offset + lensp])

    # the overall best solution may have changed. If so, replace it.
    minval = np.argmin(self.population_energies)
    if argmin:
        self.population_energies[[minval, 0]] = self.population_energies[[0, minval]]
        self.population[[0, minval], :] = self.population[[minval, 0], :]    
        if self.disp:
            print(" Best updated: f(x)= %g" % (self.population_energies[0]))
            print(self._scale_parameters(self.population[0]))